### PR TITLE
Puppet plugin fixture and tests

### PR DESCRIPTION
**Pull request by vsedmik**
This PR adds:
- fixture for puppet-enabled Satellite
- fixture for `destructive_caps` (to be registered with `destructive_sat`)
- test for puppet enable/disable logic on both, Satellite and Capsule (waiting for BZ fix)
- test fixes in `cli/test_puppetclass`, `api/test_environment`, `cli/test_environment`

TODO:
- more puppet-related test fixes will follow in next PRs

### CODE REVIEW

**Comment by vsedmik** [comment]
Test results:
```
(venv39) [vsedmik@localhost robottelo]$ pytest tests/foreman/api/test_environment.py
tests/foreman/api/test_environment.py ..............................   [100%]
=============== 30 passed, 46 warnings in 2572.97s (0:42:52) ================
```
```
(venv39) [vsedmik@localhost robottelo]$ pytest tests/foreman/cli/test_environment.py
tests/foreman/cli/test_environment.py ...........................    [100%]
=============== 27 passed, 5 warnings in 3045.71s (0:50:45) ===============
```
```
$ pytest tests/foreman/cli/test_puppetclass.py::test_positive_list_smart_class_parameters
tests/foreman/cli/test_puppetclass.py .                 [100%]
=============== 1 passed in 2932.52s (0:48:52) ===============
```

File: robottelo/hosts.py [hunk 1]
```diff
@@ -1092,6 +1093,10 @@ def install(self, installer_obj=None, cmd_args=None, cmd_kwargs=None):
             installer_obj = InstallerCommand(*cmd_args, **command_opts)
         return self.execute(installer_obj.get_command(), timeout=0)
 
+    def get_features(self):
+        """Get capsule features"""
+        return requests.get(f'https://{self.hostname}:9090/features', verify=False).text
```
Comment (tstrych) [inline robottelo/hosts.py:1098]: if this is only occurrence of using GET what about `curl -k <https..../features>`

but if we need more we need add the functionality to the Host object (if it is not already there)

File: tests/foreman/cli/test_puppetplugin.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+    result = destructive_caps.get_features()
+    assert 'puppet' not in result
+
+    result = destructive_caps.execute('rpm -q puppetserver')
+    assert result.status == 1
+    assert 'package puppetserver is not installed' in result.stdout
+
+    # Try to enable puppet on Capsule and check it failed.
+    result = destructive_caps.execute(enable_capsule_cmd.get_command(), timeout=900000)
```
Comment (tstrych) [inline tests/foreman/cli/test_puppetplugin.py:86]: please use readable format example `1h` `30m` and so on.

https://github.com/SatelliteQE/broker/blob/master/broker/helpers.py#L372-L391

File: tests/foreman/cli/test_puppetplugin.py [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+    assert result.status == 6  # == 6
+    assert 'failed to load one or more features (Puppet)' in result.stdout
+
+    # Enable puppet on Satellite and check it succeeded.
+    destructive_sat.register_to_dogfood()
+    result = destructive_sat.execute(enable_satellite_cmd.get_command(), timeout=900000)
```
Comment (tstrych) [inline tests/foreman/cli/test_puppetplugin.py:92]: same here, and from now on...

**Review by tstrych — APPROVED** [review]
ACK, 
thanks, it was really fast PR :+1: 
I think test results speak on their own 
I guess we are going down the path of duplicated fixtures.

File: tests/foreman/cli/test_puppetplugin.py [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+
+    :BZ: 2032928, 2034552, 2033336, 2039696
+    """
+    # Check that puppet is disabled by default on both.
+    assert_puppet_status(destructive_sat, expected=False)
+    assert_puppet_status(destructive_caps, expected=False)
+
+    # Try to enable puppet on Capsule and check it failed.
+    result = destructive_caps.execute(enable_capsule_cmd.get_command(), timeout='15m')
+    assert result.status == 6  # == 6
```
Comment (pondrejk) [inline tests/foreman/cli/test_puppetplugin.py:91]: this comment doesn't seem necessary

**Review by pondrejk — APPROVED** [review]

File: tests/foreman/cli/test_puppetplugin.py (hunk 4 above)
Comment (vsedmik) [reply tests/foreman/cli/test_puppetplugin.py:91]: yes, I missed that one

**Review by swadeley — APPROVED** [review]
ACK

**Pull request merged**